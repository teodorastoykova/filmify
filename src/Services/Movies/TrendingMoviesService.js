import axios from "axios";
import Movie from "../../models/Movie";
const getTrendingMovies = async () => {
  try {
    const response = await axios.get(
      "https://api.themoviedb.org/3/trending/movie/day",
      {
        params: { language: "en-US" },
        headers: {
          accept: "application/json",
          Authorization:
            "Bearer eyJhbGciOiJIUzI1NiJ9.eyJhdWQiOiI1NThkYzk1YTAxNDQyY2NmZDM0YzJlOWY0MmRiNjY5MCIsInN1YiI6IjY2MTNlOGU5MGJiMDc2MDE0ODJmYjYzNSIsInNjb3BlcyI6WyJhcGlfcmVhZCJdLCJ2ZXJzaW9uIjoxfQ.W6UGD_8V7MgsHRv2EizIyg5KV9ZZb9cHGV4A_Nq_UNY",
        },
      }
    );
    const movies = response.data.results.map((movie) => {
      return new Movie(
        movie.id,
        movie.title,
        undefined,
        movie.poster_path,
      );
    });
    return movies;
  } catch (error) {
    return error;
  }
};
export default getTrendingMovies;
