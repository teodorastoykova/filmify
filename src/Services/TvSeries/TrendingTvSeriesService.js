import axios from "axios";
import TvSeries from "../../models/TvSeries";
const getTrendingTvSeries = async () => {
  try {
    const response = await axios.get(
      "https://api.themoviedb.org/3/trending/tv/day",
      {
        params: { language: "en-US" },
        headers: {
          accept: "application/json",
          Authorization:
            "Bearer eyJhbGciOiJIUzI1NiJ9.eyJhdWQiOiI1NThkYzk1YTAxNDQyY2NmZDM0YzJlOWY0MmRiNjY5MCIsInN1YiI6IjY2MTNlOGU5MGJiMDc2MDE0ODJmYjYzNSIsInNjb3BlcyI6WyJhcGlfcmVhZCJdLCJ2ZXJzaW9uIjoxfQ.W6UGD_8V7MgsHRv2EizIyg5KV9ZZb9cHGV4A_Nq_UNY",
        },
      }
    );
    const tvSeries = response.data.results.map((tvSerie) => {
      return new TvSeries(
        tvSerie.id,
        tvSerie.name,
        undefined,
        tvSerie.poster_path,
      );
    });
    return tvSeries;
  } catch (error) {
    console.log(error);
  }
};

export default getTrendingTvSeries;
