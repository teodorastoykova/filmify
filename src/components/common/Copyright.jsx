
import { Typography } from "@mui/material";

const Copyright = (props) => {
    return (
      <Typography
        variant="body2"
        color="text.secondary"
        align="center"
        {...props}
      >
        {"Copyright Â© Filmify "}
        {new Date().getFullYear()}
        {"."}
      </Typography>
    );
  }

export default Copyright;
  