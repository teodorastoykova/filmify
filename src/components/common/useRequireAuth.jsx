/* eslint-disable react/prop-types */

const useRequireAuth = () => {
  const sessionId = localStorage.getItem('sessionId');
  return sessionId;
};

export default useRequireAuth;