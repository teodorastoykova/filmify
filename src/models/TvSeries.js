export default class TvSeries {
    constructor(id, title, overview, posterPath, releaseDate, voteAverage, seasons, episodes, productionCompanies) {
      this.id = id;
      if (title) this.title = title;
      if (overview) this.overview = overview;
      if (posterPath) this.posterPath = posterPath;
      if (releaseDate) this.releaseDate = releaseDate;
      if (voteAverage) this.voteAverage = voteAverage;
      if (seasons) this.seasons = seasons;
      if (episodes) this.episodes = episodes;
      if (productionCompanies) this.productionCompanies = productionCompanies;
    }
  }