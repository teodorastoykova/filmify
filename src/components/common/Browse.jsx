import { useEffect, useState } from "react";
import { useLocation, useNavigate } from "react-router-dom";
import axios from "axios";
import Header from "./Header";
import MoviePoster from "../movies/MoviePoster";
import TvSeriesPoster from "../tvSeries/TvSeriesPoster";
import useRequireAuth from "./useRequireAuth";

const Browse = () => {
  const sessionId = useRequireAuth();
  const navigate = useNavigate();
  const location = useLocation();
  const searchQuery = new URLSearchParams(location.search).get("query");

  const [movies, setMovies] = useState([]);
  const [tvSeries, setTvSeries] = useState([]);
  // eslint-disable-next-line no-unused-vars
  const [searchResults, setSearchResults] = useState([]);

  useEffect(() => {
    if (searchQuery) {
      const options = {
        method: "GET",
        url: "https://api.themoviedb.org/3/search/multi",
        params: {
          query: searchQuery,
          include_adult: "false",
          language: "en-US",
        },
        headers: {
          accept: "application/json",
          Authorization:
            "Bearer eyJhbGciOiJIUzI1NiJ9.eyJhdWQiOiI1NThkYzk1YTAxNDQyY2NmZDM0YzJlOWY0MmRiNjY5MCIsInN1YiI6IjY2MTNlOGU5MGJiMDc2MDE0ODJmYjYzNSIsInNjb3BlcyI6WyJhcGlfcmVhZCJdLCJ2ZXJzaW9uIjoxfQ.W6UGD_8V7MgsHRv2EizIyg5KV9ZZb9cHGV4A_Nq_UNY",
        },
      };

      axios
        .request(options)
        .then(function (response) {
          console.log(response.data);

          const results = response.data.results || [];
          setSearchResults(results);

          const moviesList = results.filter(
            (item) => item.media_type === "movie"
          );
          const tvSeriesList = results.filter(
            (item) => item.media_type === "tv"
          );

          setMovies(moviesList);
          setTvSeries(tvSeriesList);
        })
        .catch(function (error) {
          console.error(error);
        });
    }
  }, [searchQuery]);

  const handleMovieClick = (movie) => {
    navigate(`/movies/${movie.id}`);
  };

  const handleTvSeriesClick = (tvSeries) => {
    navigate(`/series/${tvSeries.id}`);
  };
  return sessionId ? (
    <div className="app-container">
      <Header />

      <h1>Search Results For {searchQuery}</h1>
      <h1>Movies</h1>
      <MoviePoster movies={movies} onMovieClick={handleMovieClick} />

      <h1>TV Series</h1>
      <TvSeriesPoster
        tvSeries={tvSeries}
        onTvSeriesClick={handleTvSeriesClick}
      />
    </div>
  ) : null;
};

export default Browse;
