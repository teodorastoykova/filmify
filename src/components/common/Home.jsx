/* eslint-disable react/prop-types */
import "../../App.css";
import { useEffect, useState } from "react";
import axios from "axios";
import Header from "./Header";
import MoviePoster from "../movies/MoviePoster";
import TvSeriesPoster from "../tvSeries/TvSeriesPoster";
import { useNavigate } from "react-router";
import useRequireAuth from "./useRequireAuth";
import { Typography } from "@mui/material";

function Home() {
  const [movies, setMovies] = useState([]);
  const [tvSeries, setTvSeries] = useState([]);

  const navigate = useNavigate();
  const sessionId = useRequireAuth();

  useEffect(() => {
    if (sessionId) {
      const fetchTrendingData = async () => {
        try {
          const trendingMoviesResponse = await axios.get(
            "https://api.themoviedb.org/3/trending/movie/day",
            {
              params: { language: "en-US" },
              headers: {
                accept: "application/json",
                Authorization:
                  "Bearer eyJhbGciOiJIUzI1NiJ9.eyJhdWQiOiI1NThkYzk1YTAxNDQyY2NmZDM0YzJlOWY0MmRiNjY5MCIsInN1YiI6IjY2MTNlOGU5MGJiMDc2MDE0ODJmYjYzNSIsInNjb3BlcyI6WyJhcGlfcmVhZCJdLCJ2ZXJzaW9uIjoxfQ.W6UGD_8V7MgsHRv2EizIyg5KV9ZZb9cHGV4A_Nq_UNY",
              },
            }
          );

          const trendingTvResponse = await axios.get(
            "https://api.themoviedb.org/3/trending/tv/day",
            {
              params: { language: "en-US" },
              headers: {
                accept: "application/json",
                Authorization:
                  "Bearer eyJhbGciOiJIUzI1NiJ9.eyJhdWQiOiI1NThkYzk1YTAxNDQyY2NmZDM0YzJlOWY0MmRiNjY5MCIsInN1YiI6IjY2MTNlOGU5MGJiMDc2MDE0ODJmYjYzNSIsInNjb3BlcyI6WyJhcGlfcmVhZCJdLCJ2ZXJzaW9uIjoxfQ.W6UGD_8V7MgsHRv2EizIyg5KV9ZZb9cHGV4A_Nq_UNY",
              },
            }
          );
          setMovies(trendingMoviesResponse.data.results);
          setTvSeries(trendingTvResponse.data.results);
        } catch (error) {
          console.error("Error fetching trending data:", error);
        }
      };
      fetchTrendingData();
    } else {
      navigate("/login");
    }
  }, [navigate, sessionId]);

  const handleMovieClick = (movie) => {
    navigate(`/movies/${movie.id}`);
  };

  const handleTvSeriesClick = (tvSeries) => {
    navigate(`/series/${tvSeries.id}`);
  };

  return sessionId ? (
    <>
      <Header />
      <Typography variant="h3" gutterBottom>
        Trending Movies
      </Typography>
      <MoviePoster movies={movies} onMovieClick={handleMovieClick} />
      <Typography variant="h3" gutterBottom>
        Trending TV Series
      </Typography>
      <TvSeriesPoster
        tvSeries={tvSeries}
        onTvSeriesClick={handleTvSeriesClick}
      />
    </>
  ) : null;
}

export default Home;
