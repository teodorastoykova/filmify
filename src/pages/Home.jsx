/* eslint-disable react/prop-types */
import "../App.css";
import { useEffect, useState } from "react";
import Header from "../components/common/Header";
import MoviePoster from "../components/movies/MoviePoster";
import TvSeriesPoster from "../components/tvSeries/TvSeriesPoster";
import { useNavigate } from "react-router";
import useRequireAuth from "../components/common/useRequireAuth";
import { Typography } from "@mui/material";
import getTrendingMovies from "../services/Movies/TrendingMoviesService";
import getTrendingTvSeries from "../services/TvSeries/TrendingTvSeriesService";

function Home() {
  const [movies, setMovies] = useState([]);
  const [tvSeries, setTvSeries] = useState([]);

  const navigate = useNavigate();
  const sessionId = useRequireAuth();

  useEffect(() => {
    if (sessionId) {
      const fetchTrendingData = async () => {
        try {
          const trendingMoviesResponse = await getTrendingMovies();
          const trendingTvResponse = await getTrendingTvSeries();
          setMovies(trendingMoviesResponse);
          setTvSeries(trendingTvResponse);
        } catch (error) {
          console.error("Error fetching trending data:", error);
        }
      };
      fetchTrendingData();
    } else {
      navigate("/login");
    }
  }, [navigate, sessionId]);

  const handleMovieClick = (movie) => {
    navigate(`/movies/${movie.id}`);
  };

  const handleTvSeriesClick = (tvSeries) => {
    navigate(`/series/${tvSeries.id}`);
  };

  return sessionId ? (
    <>
      <Header />
      <Typography variant="h3" gutterBottom>
        Trending Movies
      </Typography>
      <MoviePoster movies={movies} onMovieClick={handleMovieClick} />
      <Typography variant="h3" gutterBottom>
        Trending TV Series
      </Typography>
      <TvSeriesPoster
        tvSeries={tvSeries}
        onTvSeriesClick={handleTvSeriesClick}
      />
    </>
  ) : null;
}

export default Home;
