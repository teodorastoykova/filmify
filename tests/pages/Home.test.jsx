import { beforeEach, describe, expect, it, vi } from "vitest";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import Home from "../../src/pages/Home";
import * as getTrendingMovies from "../../src/services/Movies/TrendingMoviesService";
import * as getTrendingTvSeries from "../../src/services/TvSeries/TrendingTvSeriesService";
import * as useRequireAuth from "../../src/components/common/useRequireAuth";
import * as router from "react-router";

describe("Home page", () => {
  const mockGetTrendingMovies = vi.fn();
  const mockGetTrendingTvSeries = vi.fn();
  const mockUseRequireAuth = vi.fn();
  const mockUseNavigate = vi.fn();

  const mockedMovies = [
    { id: 1, title: "Movie 1", posterPath: "/poster1.jpg" },
    { id: 2, title: "Movie 2", posterPath: "/poster2.jpg" },
    { id: 3, title: "Movie 3", posterPath: "/poster3.jpg" },
    { id: 4, title: "Movie 4", posterPath: "/poster4.jpg" },
    { id: 5, title: "Movie 5", posterPath: "/poster5.jpg" },
    { id: 6, title: "Movie 6", posterPath: "/poster6.jpg" },
  ];

  const mockedTvSeries = [
    { id: 1, title: "TvSeries 1", posterPath: "/poster1.jpg" },
    { id: 2, title: "TvSeries 2", posterPath: "/poster2.jpg" },
    { id: 3, title: "TvSeries 3", posterPath: "/poster3.jpg" },
    { id: 4, title: "TvSeries 4", posterPath: "/poster4.jpg" },
    { id: 5, title: "TvSeries 5", posterPath: "/poster5.jpg" },
    { id: 6, title: "TvSeries 6", posterPath: "/poster6.jpg" },
  ];
  beforeEach(() => {
    vi.spyOn(getTrendingMovies, "default").mockImplementation(
      mockGetTrendingMovies
    );
    vi.spyOn(getTrendingTvSeries, "default").mockImplementation(
      mockGetTrendingTvSeries
    );
    vi.spyOn(useRequireAuth, "default").mockImplementation(mockUseRequireAuth);
  });

  it("renders a snapshot", () => {
    mockGetTrendingMovies.mockResolvedValue(mockedMovies);
    mockGetTrendingTvSeries.mockResolvedValue(mockedTvSeries);
    mockUseRequireAuth.mockResolvedValue(1);
    const { container } = render(
      <MemoryRouter>
        <Home />
      </MemoryRouter>
    );

    expect(container.firstChild).toMatchSnapshot();
  });
  it("fetches trending movies and TV series data when user is authenticated", async () => {
    mockGetTrendingMovies.mockResolvedValue(mockedMovies);
    mockGetTrendingTvSeries.mockResolvedValue(mockedTvSeries);
    mockUseRequireAuth.mockResolvedValue(1);

    const { getByText } = render(
      <MemoryRouter>
        <Home />
      </MemoryRouter>
    );

    expect(getByText("Trending Movies")).toBeInTheDocument();
    expect(getByText("Trending TV Series")).toBeInTheDocument();

    expect(mockGetTrendingMovies).toHaveBeenCalled();
    expect(mockGetTrendingTvSeries).toHaveBeenCalled();
  });
  it("redirects to login page when user is not authenticated", async () => {
    vi.spyOn(router, "useNavigate").mockImplementation(() => mockUseNavigate);

    mockUseRequireAuth.mockReturnValue(null);
    const { queryByText } = render(
      <MemoryRouter>
        <Home />
      </MemoryRouter>
    );

    expect(queryByText("Trending Movies")).not.toBeInTheDocument();
    expect(queryByText("Trending TV Series")).not.toBeInTheDocument();
    expect(mockUseNavigate).toHaveBeenCalledWith("/login");
  });
});
