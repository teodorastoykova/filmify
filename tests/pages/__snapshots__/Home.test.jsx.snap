// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Home page > Renders a snapshot 1`] = `
<div
  class="MuiGrid-root MuiGrid-container css-1oukqxz-MuiGrid-root"
>
  <div
    class="MuiGrid-root MuiGrid-item title css-13i4rnv-MuiGrid-root"
  >
    Filmify
  </div>
  <div
    class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1e6y48t-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Home
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1e6y48t-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Top Rated
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1e6y48t-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      My Watch List
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1e6y48t-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Log Out
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
  <div
    class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
  />
  <div
    class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
  >
    <button
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="SearchIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="NotificationsNoneIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 22c1.1 0 2-.9 2-2h-4c0 1.1.9 2 2 2m6-6v-5c0-3.07-1.63-5.64-4.5-6.32V4c0-.83-.67-1.5-1.5-1.5s-1.5.67-1.5 1.5v.68C7.64 5.36 6 7.92 6 11v5l-2 2v1h16v-1zm-2 1H8v-6c0-2.48 1.51-4.5 4-4.5s4 2.02 4 4.5z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;

exports[`Home page > renders a snapshot 1`] = `
<div
  class="MuiGrid-root MuiGrid-container css-1oukqxz-MuiGrid-root"
>
  <div
    class="MuiGrid-root MuiGrid-item title css-13i4rnv-MuiGrid-root"
  >
    Filmify
  </div>
  <div
    class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1e6y48t-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Home
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1e6y48t-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Top Rated
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1e6y48t-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      My Watch List
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1e6y48t-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Log Out
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
  <div
    class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
  />
  <div
    class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
  >
    <button
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="SearchIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="NotificationsNoneIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 22c1.1 0 2-.9 2-2h-4c0 1.1.9 2 2 2m6-6v-5c0-3.07-1.63-5.64-4.5-6.32V4c0-.83-.67-1.5-1.5-1.5s-1.5.67-1.5 1.5v.68C7.64 5.36 6 7.92 6 11v5l-2 2v1h16v-1zm-2 1H8v-6c0-2.48 1.51-4.5 4-4.5s4 2.02 4 4.5z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
